packages/react-devtools-shared/src/hydration.js

```diff
@@ -151,7 +151,10 @@ export function dehydrate(
         inspectable: false,
         preview_short: formatDataForPreview(data, false),
         preview_long: formatDataForPreview(data, true),
-        name: data.name || 'function',
+        name:
+          typeof data.name === 'function' || !data.name
+            ? 'function'
+            : data.name,
```

Comment:
The javascript Proxy wraps the name attribute. I think instead of using `data.name` you can replace  it by `data.prototype.constructor.name`. It is the same thing but the difference is the constructor.name is not wrapped by the proxy.

Reply:
> The javascript Proxy wraps the name attribute.

Oh that's very interesting! Nice find.

> I think instead of using `data.name` you can replace  it by `data.prototype.constructor.name`. It is the same thing but the difference is the constructor.name is not wrapped by the proxy.

I'm wary of using `data.prototype.constructor.name` because not all objects have a prototype (e.g. `Object.create(null)`